R/dimensionreductionscatterplot.R

```diff
@@ -275,193 +275,71 @@ convertFactorAnalysisTo2D <- function(x) {
+                                 y.title = "Dimension 2",
+                                 labels.font.size = 14,
+                                 x.title.font.size = 14,
+                                 y.title.font.size = 14,
+                                 point.radius = 4,
+                                 fixed.aspect = TRUE,
+                                 plot.border.show = TRUE,
+                                 origin = TRUE)
     }
 
     else {
         # Scatterplot with groups
         scatter.group.indices <- ""
         scatter.group.labels <- ""
-        legend <- TRUE
-        colors <- "Default colors"
+        colors <- ChartColors(12)
         title <- x$title
         set.seed(1066)
 
         embedding <- x$embedding[x$used.subset, ]
         groups <- x$data.groups[x$used.subset]
 
+        scatter.colors <- NULL
         if (!is.null(groups)) {
-
             if (is.factor(groups)) {
-                scatter.group.indices <- paste(as.numeric(groups), collapse = ", ")
-                scatter.group.labels <- paste(levels(groups), collapse = ", ")
+                scatter.colors <- groups
                 nearest <- knn.cv(train = embedding, cl = groups, k = 1)
                 same.category <- Sum(nearest == factor(groups, ordered = FALSE), remove.missing = FALSE) / length(groups)
                 title <- paste0(title, " - Nearest neighbor accuracy: ", sprintf("%1.2f%%", 100 * same.category))
             }
             else if (all(groups == floor(groups))) {
-                unique.labels <- sort(unique(groups))
-                indices <- match(groups, unique.labels)
-                scatter.group.labels <- paste(unique.labels, collapse = ", ")
-                scatter.group.indices <- paste(indices, collapse = ", ")
-                colors <- "Reds, light to dark"
+                scatter.colors <- factor(groups)
+                colors <- ChartColors(length(unique(scatter.colors)), "Reds, light to dark")
                 nearest <- knn.cv(train = embedding, cl = groups, k = 1)
                 same.category <- Sum(nearest == groups, remove.missing = FALSE) / length(groups)
                 title <- paste0(title, ". Nearest neighbor accuracy: ", sprintf("%1.2f%%", 100 * same.category))
             }
             else {       # numeric: create 20 buckets and treat as factors
-                groups <- cut(groups, 20)
-                scatter.group.indices <- paste(as.numeric(groups), collapse = ", ")
-                levels(groups) <- sub("[^,]*,([^]]*)\\]", "\\1", levels(groups))
-                scatter.group.labels <- paste(levels(groups), collapse = ", ")
-                colors <- "Reds, light to dark"
+                scatter.colors <- cut(groups, 20)
+                levels(scatter.colors) <- sub("[^,]*,([^]]*)\\]", "\\1", levels(scatter.colors))
+                colors <- ChartColors(20, "Reds, light to dark")
             }
         }
 
-        chart <- Chart(y = embedding,
```

Comment:
Not sure why we needed to pass parameters with their default values. I have removed the unnecessary ones.